### PR TITLE
Support generic stream/event on XPU backend

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #125757
* __->__ #125751

# Motivation
According to [#123611](https://github.com/pytorch/pytorch/pull/123611), we support generic stream/event on XPU backend.

# Additional Context
new method/attribute on `torch.Event` for xpu
- torch.Event.event_id
- torch.Event.elapsed_time
- torch.Event.synchronize

new method on `c10::Event` on xpu backend
- c10.Event.event_id
- c10.Event.elapsed_time
- c10.Event.synchronize

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @EikanWang @fengyuan14

### CODE REVIEW

File: test/test_cpp_extensions_mtia_backend.py [hunk 1]
```diff
@@ -34,7 +35,7 @@ def remove_build_path():
 
 
 @unittest.skipIf(
-    IS_ARM64 or not IS_LINUX or TEST_CUDA or TEST_PRIVATEUSE1 or TEST_ROCM,
+    IS_ARM64 or not IS_LINUX or TEST_CUDA or TEST_PRIVATEUSE1 or TEST_ROCM or TEST_XPU,
```
Comment (guangyey) [inline test/test_cpp_extensions_mtia_backend.py:38]: It is mutually exclusive to xpu.

**Comment by guangyey** [comment]
@albanD, if this PR is fine, I will continue to support CUDA following this.

**Review by jgong5 — APPROVED** [review]

File: test/test_cpp_extensions_mtia_backend.py (hunk 1 above)
Comment (albanD) [reply test/test_cpp_extensions_mtia_backend.py:38]: cc @egienvalue with me want to add a TEST_MTIA to common_utils above to simplify this kind of statement!

**Review by albanD — APPROVED** [review]
Very cool!

File: test/test_cpp_extensions_mtia_backend.py (hunk 1 above)
Comment (egienvalue) [reply test/test_cpp_extensions_mtia_backend.py:38]: I see. Thanks for the suggestion. I will do that.

**Comment by guangyey** [comment]
> Very cool!

Thanks for your approval~

**Pull request closed without merging**